python/pyarrow/array.pxi

```diff
@@ -149,12 +149,16 @@ def array(object obj, type=None, mask=None, size=None, bint from_pandas=False,
     """
     type = ensure_type(type, allow_none=True)
     cdef CMemoryPool* pool = maybe_unbox_memory_pool(memory_pool)
+    cdef bint is_pandas_object = False
 
     if _is_array_like(obj):
         if mask is not None:
-            mask = get_series_values(mask)
+            # out argument unused
+            mask = get_series_values(mask, &is_pandas_object)
 
-        values = get_series_values(obj)
+        values = get_series_values(obj, &is_pandas_object)
+        if is_pandas_object:
+            from_pandas = True
```

Comment:
If the user passes `pandas.Series` to `pyarrow.array`, they probably mean `from_pandas=True=`, would you all agree?

Reply:
Sounds ok to me.

Reply:
I would only update the docstring then to mention this.

Reply:
The default could also be change to None, leaving the option to user to force True or False, regardless the container type (but not sure that is worth it)

Reply:
I thought about that (making the default None). I can go ahead and make that change